pygeos/io.py

```diff
@@ -219,33 +296,47 @@ def from_shapely(geometry, **kwargs):
     >>> from shapely.geometry import Point   # doctest: +SKIP
     >>> from_shapely(Point(1, 2))   # doctest: +SKIP
     <pygeos.Geometry POINT (1 2)>
+
+    Notes
+    -----
+    If PyGEOS and Shapely do not use the same GEOS version,
+    the conversion happens through the WKB format and will thus be slower.
     """
     check_shapely_version()
-
-    if shapely_geos_version is None:
+    if shapely_compatible is None:
         raise ImportError("This function requires shapely")
 
-    # shapely has something like: "3.6.2-CAPI-1.10.2 4d2925d6"
-    # pygeos has something like: "3.6.2-CAPI-1.10.2"
-    if not shapely_geos_version.startswith(geos_capi_version_string):
-        raise ImportError(
-            "The shapely GEOS version ({}) is incompatible with the GEOS "
-            "version PyGEOS was compiled with ({})".format(
-                shapely_geos_version, geos_capi_version_string
-            )
-        )
-
-    if isinstance(geometry, ShapelyGeometry):
-        # this so that the __array_interface__ of the shapely geometry is not
-        # used, converting the Geometry to its coordinates
-        arr = np.empty(1, dtype=object)
-        arr[0] = geometry
-        arr.shape = ()
-    elif not isinstance(geometry, np.ndarray) and isinstance(geometry, Sized):
-        # geometry is a list/array-like
-        arr = np.empty(len(geometry), dtype=object)
-        arr[:] = geometry
+    if shapely_compatible:
+        if isinstance(geometry, ShapelyGeometry):
```

Comment:
Have you tested what happens if `geometry`  is a shapely `PreparedGeometry` ? See https://github.com/pygeos/pygeos/issues/119

Reply:
I got it working for the case when the GEOS versions do not match, but I did not change anything to the existing C code.

I can look into this, but wanted some pointers on your end.
- in [ufunc.c](https://github.com/pygeos/pygeos/blob/master/src/ufuncs.c#L2454), I can check whether the object has a `context` attribute and use `obj.context.__geom__` in that case.
- Is there a better approach in C ? (eg. like `isinstance(obj, PreparedGeometry)`)